synapse/http/endpoint.py

```diff
@@ -154,6 +156,12 @@ def connect(self, protocolFactory):
 
 @defer.inlineCallbacks
 def resolve_service(service_name, dns_client=client, cache=SERVER_CACHE):
+    cache_entry = cache.get(service_name, None)
+    if cache_entry:
+        if all(s.expires > int(time.time()) for s in cache_entry):
```

Comment:
Hmm, I tend to prefer to pass in the current time to these things to make testing easier.